src/pages/Alerts/Alerts.js

```diff
@@ -0,0 +1,117 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import Loadable from 'react-loadable'
+import { useLocation } from 'react-router-dom'
+import { parse } from 'query-string'
+import cx from 'classnames'
+import withSizes from 'react-sizes'
+import AlertsFilter, { filters } from './AlertsFilter/AlertsFilter'
+import IndexTab from '../../components/IndexTabs/IndexTab'
+import PageLoader from '../../components/Loader/PageLoader'
+import { MobileOnly } from '../../components/Responsive'
+import { RecommendedSignals } from '../SonarFeed/SonarFeedRecommendations'
+import { SignalModal } from '../SonarFeed/SonarFeedPage'
+import { useSignals } from '../../ducks/Signals/common/getSignals'
+import { useUser } from '../../stores/user'
+import { mapSizesToProps } from '../../utils/withSizes'
+import styles from './Alerts.module.scss'
+
+const LoadableAlertsList = Loadable({
+  loader: () => import('../SonarFeed/SignalsList'),
+  loading: () => <PageLoader />
+})
+
+const Alerts = ({ isDesktop }) => {
+  const [filter, setFilter] = useState(filters.ALL)
+  const { user, loading: isUserLoading } = useUser()
+  const { tab } = parse(useLocation().search, { parseNumbers: true })
+  const { data: signals = [], loading } = useSignals({
+    skip: user && !user.id
+  })
+  const initialTab = useMemo(() => {
+    if (tab) {
+      return tab
+    }
+
+    return signals && signals.length > 0 ? 1 : 0
+  }, [signals, tab])
```

Comment:
There is no complex computations, hence we can avoid the overhead and do a regular check: 
```js
tab || (signals && signals.length > 0 ? 1 : 0)
```